### ISSUE TITLE
Support building for dev/staging sites.

### ISSUE THREAD

**Issue by jaunt** [gridsome/gridsome#720]
## Summary

I use Vue Cli to build to my dev, staging, and production sites like this:

```
vue-cli-service build --mode development
vue-cli-service build --mode staging
vue-cli-service build
```

This causes .env files to be loaded respectively:

```
.env.development
.env.staging
.env.producton
```

This allows me to reference other staging APIs, have different base URLs, have debug and logging features enabled, different analytics projects, as well as SEO suppression.

### Basic example

gridsome build staging

### Motivation

Having a live dev and staging site has been critical for my team.  It allows remote beta testing etc, and improves our quality dramatically.

**Comment by hjvedvik** [comment]
Yes, that would be nice to have. I'll add this to the todo-list :)

**Comment by jaunt** [comment]
Awesome!  Thanks

**Comment by MWSolutions** [comment]
thanks, I'm also looking for this feature

**Comment by jaunt** [comment]
My current workaround, fwiw.

1.  I have a folder called "environments" within the project.  Within I have folders dev, local, staging, and production.  In each of those I have an .env.production file and a folder called src with a file index.html.  

2.   My package.json has scripts like so:

`    "scripts": {
    "build-dev": "yarn run env-dev && gridsome build",
    "build-staging": "yarn run env-staging && gridsome build",
    "build-production": "yarn run env-production && gridsome build",
    "env-dev": "cp -r ./environments/dev/. .",
    "env-production": "cp -r ./environments/production/. .",
    "env-staging": "cp -r ./environments/staging/. .",
    "env-local": "cp -r ./environments/local/. .",
    "develop": "yarn run env-local && gridsome develop",
    "explore": "gridsome explore"
  }`

The index.html file I use to install 3rd party scripts that can't be done with vue-meta, and it's where I have different google tag manager environments.  env.production has GRIDSOME_APP_DEPLOY_SITE=staging/production etc.

**Comment by michaelmoneypenny** [comment]
Is this still the same? I have the exact same requirement. @jaunt your example looks like a great workaround!

**Comment by derrickdso** [comment]
Also looking for this feature to be added!

**Comment by chrisrueger** [comment]
I have extended the idea above a little bit so that I can **build** and **develop** for against two different environments. 


```
# package.json

"scripts": {
    "env-build-production": "cp -r ./environments/production/. .",
    "env-build-local": "cp -r ./environments/local/. .",
    "env-develop-production": "cp ./environments/production/.env.production .env",
    "env-develop-local": "cp ./environments/local/.env.production .env",
    "build-local": "yarn run env-build-local && gridsome build",
    "build": "yarn run env-build-production && gridsome build",
    "develop": "yarn run env-develop-local && gridsome develop",
    "develop-production": "yarn run env-develop-production && gridsome develop",
    "explore": "gridsome explore"
  },

```

Make sure you add to `.gitignore`

```
.env
.env.production
```

`environment/local/.env.production` contains e.g.:

```
MY_API_URL=http://localhost
```

This allows me to do:

`yarn develop` starts dev-server and uses http://localhost to fetch API data via `gridsome-source-rest`
`yarn develop-production` starts dev-server and uses https://api.example.com to fetch API data via `gridsome-source-rest`

For build it is similar but the other way around:

`yarn build` uses `production` https://api.example.com to fetch API data via `gridsome-source-rest` and creates the `/dist` folder
`yarn build-local` uses http://localhost to fetch API data via `gridsome-source-rest` and creates the `/dist` folder


Would be cool if this could be built-in to the main commands.
